File: 프로그래머스/lv2/12973. 짝지어 제거하기/짝지어 제거하기.js
function solution(s) {
    let stack = [];
    for (let i of s) {
        stack.push(i);
        if (stack[stack.length - 2] === stack[stack.length - 1]) {
            stack.pop();
            stack.pop();
        }
    }
    if (stack.length === 0) return 1;
    else return 0;
}

File: 프로그래머스/2/12973. 짝지어 제거하기/짝지어 제거하기.py
def solution(s):
    stk = []
    
    for i in s:
        stk.append(i)
        if len(stk) > 1 and stk[-1] == stk[-2]:
            stk.pop()
            stk.pop()
    if len(stk) == 0:
        return 1
    return 0